mapper/pojo-base/src/main/java/org/hibernate/search/mapper/pojo/bridge/binding/impl/AbstractBindingContext.java

```diff
@@ -35,22 +36,18 @@ public BeanResolver beanResolver() {
 
 	@Override
 	public <T> T param(String name, Class<T> paramType) {
-		Contracts.assertNotNull( name, "name" );
-		Contracts.assertNotNull( paramType, "paramType" );
-
-		Object value = params.get( name );
-		if ( value == null ) {
-			throw log.paramNotDefined( name );
-		}
-
-		return paramType.cast( value );
+		return params.get( name, paramType );
 	}
 
 	@Override
 	public <T> Optional<T> paramOptional(String name, Class<T> paramType) {
```

Comment:
Shouldn't older methods get deprecated?

Reply:
well that backfired nicely 😅 
```
Error:  Constraint: hsearch:TypesExtendingTypeFromAnotherModuleMustHaveModuleSpecificKeywordInName
Error:  Severity: MAJOR
Error:  Number of rows: 22
Error:  
Error:  Main (non-test) types extending/implementing a Hibernate Search type from another module must have a module-specific keyword in their name,
Error:  either at the very start or just after "Abstract".
Error:  This allows to more easily understand which module a given type comes from.
Error:  Exceptions are allowed when:
Error:  - the misnamed type is an anonymous type or a generated type;
Error:  - or the misnamed type is deprecated;
Error:  - or the implemented type is deprecated;
Error:  - or the implementing type is an inner/nested type and is non-public (implementation) --
Error:  in which case only the surrounding type must comply, if relevant;
Error:  - or the implemented type is in a util module,
Error:  in which case the implemented interface may just be a detail.
Error:  
Error:  misnamedTypeArtifact.moduleSpecificKeyword=Pojo, misnamedType=org.hibernate.search.mapper.pojo.bridge.binding.impl.PropertyBindingContextImpl, externalParentType=org.hibernate.search.engine.search.common.NamedValues
```
Once I made `BindingContext extends NamedValues` 😄

Reply:
Yeah, don't do that :)